pymeasure/instruments/siglenttechnologies/siglent_spdbase.py

```diff
@@ -0,0 +1,294 @@
+        self.recall_settings = index
+
+
+class SPDChannel(object):
+    """ The channel class for Siglent SPDxxxxX instruments.
+    """
+
+    def __init__(self, instrument, channel: int = 1):
+        self.instrument = instrument
+        self.channel = channel
+
+    voltage = Instrument.measurement(
+        "MEAS:VOLT? CH{channel}",
+        """Measure the channel output voltage.
+
+        :type: float
+        """
+    )
+
+    current = Instrument.measurement(
+        "MEAS:CURR? CH{channel}",
+        """Measure the channel output voltage.
+
+        :type: float
+        """
+    )
+
+    power = Instrument.measurement(
+        "MEAS:POWE? CH{channel}",
+        """Measure the channel output voltage.
+
+        :type: float
+        """
+    )
+
+    set_current = Instrument.control(
+        "CH{channel}:CURR?", "CH{channel}:CURR %g",
+        """Control the output current configuration of the channel.
+
+        :type: float
+
+        """,
+        validator=truncated_range,
+        values=[0, 8],
+        dynamic=True
+    )
+
+    set_voltage = Instrument.control(
+        "CH{channel}:VOLT?", "CH{channel}:VOLT %g",
+        """Control the output voltage configuration of the channel.
+
+        :type: float
+
+        """,
+        validator=truncated_range,
+        values=[0, 16],
+        dynamic=True
+    )
+
+    output = Instrument.setting(
+        "OUTP CH{channel},%s",
+        """Configure the power supply output.
+
+        :type: bool
+            ``True``: enables the output
+            ``False``: disables the output
+
+        """,
+        validator=strict_discrete_set,
+        values={True: "ON", False: "OFF"},
+        map_values=True
+    )
+
+    enable_timer = Instrument.setting(
+        "TIME CH{channel},%s",
+        """Enable the channel timer.
+
+        :type: bool
+            ``True``: enables the timer
+            ``False``: disables it
+
+        """,
+        validator=strict_discrete_set,
+        values={True: "ON", False: "OFF"},
+        map_values=True
+    )
+
+    def ask(self, cmd):
+        return self.instrument.ask(cmd.format(channel=self.channel))
+
+    def write(self, cmd):
+        self.instrument.write(cmd.format(channel=self.channel))
+
+    def values(self, cmd, **kwargs):
+        """ Reads a set of values from the instrument through the adapter,
+        passing on any key-word arguments.
+        """
+        return self.instrument.values(cmd.format(channel=self.channel), **kwargs)
+
+    def check_errors(self):
+        return self.instrument.check_errors()
+
+    def enable(self):
+        """Enable the channel output
+
+        :returns: self
+        """
+        self.instrument.selected_channel = self.channel
+        self.output = True
+        return self
+
+    def disable(self):
+        """Disable the channel output
+
+        :returns: self
+        """
+        self.instrument.selected_channel = self.channel
+        self.output = False
+        return self
```

Comment:
Why does this duplicate what `output` does? 
Is that the only function that needs that channel selection machinery? If yes, would using the channels `.output` alone even work correctly?
Why are you returning `self` here?

Reply:
I've changed the output part to remove property and replace with an `enable_output` method. hopefully this will make more sense overall